src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/PoolingAsyncValueTaskMethodBuilderT.cs

```diff
@@ -403,7 +403,7 @@ public void MoveNext()
             {
                 ExecutionContext? context = Context;
 
-                if (context is null)
+                if (context == ExecutionContext.DefaultFlowSuppressed)
```

Comment:
Are there still cases where context is null? If so, those will now end up taking the more expensive ExecutionContext.RunInternal path?